### PR TITLE
docs(specs): discipline-layer migration from raw bd to the work facade

### PR DESCRIPTION
**Pull request by scotthamilton77**
Design spec for migrating every bd-calling discipline-layer asset under src/** onto the `work` facade (fablize batch, bead agents-config-wgclw.9.4 — promoted to an epic by this spec).

- **Central artifact: a 24-file mapping table** — every bd reference classified route-through-facade / stays-bd-specific / facade-gap, with target verbs per file.
- **Gap register with in-session verification verdicts:** G1 (typed dep edges) and G2 (multi-status list filter) verified ALREADY COVERED against the shipped code; G3 (claim/started timestamp) confirmed a real gap — one additive `started` envelope field, MINOR bump.
- **Partition:** five child slices (A programmatic skills → B PR-workflow skills → C triage/create skills, D/E audit-only in parallel). Class C's filing recipe targets `work discover` (spec ships in a sibling PR) — never raw create+dep-add, which would reinstate the unanchored-filing leak.
- **Rollout:** everything gated behind installer CLI-deploy (wgclw.9.9); migrated assets fail loud when `work` is absent from PATH — no silent bd fallback.

Spec only — no implementation. Continuations name the epic's child slices + the G3 field. Reviewed: frontier fact-check + quality-reviewer pass, all findings applied.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01SYiw9n199ChDS3HvwFw5mp

### CODE REVIEW

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+(`packages/workcli`) exists precisely to quarantine bd behind one tested JSON-envelope
+contract, but no installed asset consumes it yet. This epic migrates every asset that
+*invokes* bd onto the facade, so assets depend on the stable contract, not the bd
+implementation — landing the M0 charter's "beads is quarantined behind our own CLI"
+all the way to the asset surface.
+
+Facade verb surface migrated against — transport (contract spec §3): `show`, `create --raw`,
+`update`, `note`, `close`, `reopen`, `list`, `ready`, `dep`, `label`, `search`, `sync`;
+lifecycle (lifecycle plan): `create <noun>`, `claim`, `release`, `deliver`, `plan`,
+`promote`, `reconcile`. A bd usage maps only if one of these covers it.
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in 443d18906e88b5e893f9e0778b1589021fbbb455. Replaced the vague labels with the concrete documents: docs/specs/2026-07-04-work-facade-cli-contract.md §3 and docs/plans/2026-07-12-workcli-lifecycle-layer.md.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+## 2. Decision summary (normative — do not reopen)
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in 0b3b548d76e7f3bc599b80764c2fc3e3267401aa. Reworded the heading to "Decision summary (owner-ratified)" and added a clarifying note: decisions were owner-ratified pre-draft, bind implementation, and remain open to review comments.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  9.9 makes PATH-reachability a *deployment invariant* (its §6), so steady state has
+  `work` present; the guards below cover partial/broken installs only.
+- **`work` absent from PATH → fail loud.** A migrated asset that cannot find `work` must
+  emit a not-configured error and stop — **never silently fall back to `bd`** (that would
+  re-introduce the drift the facade removes). Prose skills: a one-line guard ("if `work`
+  is not on PATH, stop and tell the operator to run the installer — do not use raw `bd`").
+  `collect.py`: on `FileNotFoundError` for `work`, exit non-zero with a
+  `work-not-installed` error naming the installer (deliberately NOT echoing a real
+  `ErrorCode` member — this fires in the calling script before `work` is ever invoked), not the empty-list swallow it
+  uses today for bd absence.
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in dce1e2c496f4b7d90525980b0b38389283a062d5. Split the bullet into three sentences preserving the non-zero exit, work-not-installed error, ErrorCode caveat, and replacement of the empty-list swallow.

**Comment by scotthamilton77** [comment]
Overview summary only — accurately describes the spec; no changes requested, no action needed.

**Comment by scotthamilton77** [comment]
@codex review

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+| `src/user/.agents/skills/merge-guard/SKILL.md` | `label list <id> --json \| jq join` | RTF | `work label list` (envelope `string[]`, no jq) | B |
+| `src/user/.agents/skills/wait-for-pr-comments/SKILL.md` | `label add <id> human`, `update --append-notes`, `label list --json`; `.beads/worker-audit/` path | RTF (path=BDS) | `work label add`, `work note`, `work label list`; `.beads/` audit path stays bd infra | B |
+| `src/user/.agents/skills/reply-and-resolve-pr-threads/SKILL.md` | `label add <id> human` + `update --append-notes`; "no bd jargon in replies" policy | RTF (policy=BDS) | `work label add`, `work note`; PR-hygiene policy unaffected (output rule, not a call) | B |
+| `src/user/.agents/skills/triaging-discovered-work/SKILL.md` | `create --parent`, `dep add --type discovered-from`, `reopen`, `create` | RTF | the filing recipe targets **`work discover`** (the composed verb — see the 2026-07-17 work-discover-verb spec), NOT raw `work create` + `work dep add`, which would reinstate the unanchored-filing leak the verb closes; `work reopen` and reads migrate directly (typed dep edges verified shipped end-to-end if ever needed raw) | C |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md:64]: The work-discover verb spec ships in sibling PR #301 (docs/specs/2026-07-17-work-discover-verb.md), opened alongside this PR — the reference resolves once both merge, and the Class C slice is explicitly sequenced behind that verb landing, so the filing-path contract is concretely tracked rather than missing.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  `filters.status` verbatim to `bd list --status`, which accepts the comma list
+  (`open,in_progress`) `collect.py` uses today. Pin a contract test in the Class A
+  child rather than filing a verb.
+- **G3 — claim/started timestamp: CONFIRMED REAL GAP.** The `Item` model carries
+  `created`/`updated` but no claim/started timestamp; `collect.py`'s in-flight
+  detection reads bd's `started_at`. Resolution: an additive `started` field on the
+  item envelope shape sourced from bd's `started_at` (`None` when unset), a MINOR
+  protocol bump. Filed as a continuation.
+
+G3 is the only real gap, and it is an additive envelope field, not a verb. Formulas
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in 981abbcb53c57d34a1bf6f1e0e4890b0138535ce. Widened the G3 gap register to two additive Item envelope fields (started AND assignee) with the type/created remapping note for Class A; partition table, acceptance criterion 6, and the continuation updated.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+| `src/user/.agents/skills/reply-and-resolve-pr-threads/SKILL.md` | `label add <id> human` + `update --append-notes`; "no bd jargon in replies" policy | RTF (policy=BDS) | `work label add`, `work note`; PR-hygiene policy unaffected (output rule, not a call) | B |
+| `src/user/.agents/skills/triaging-discovered-work/SKILL.md` | `create --parent`, `dep add --type discovered-from`, `reopen`, `create` | RTF | the filing recipe targets **`work discover`** (the composed verb — see the 2026-07-17 work-discover-verb spec), NOT raw `work create` + `work dep add`, which would reinstate the unanchored-filing leak the verb closes; `work reopen` and reads migrate directly (typed dep edges verified shipped end-to-end if ever needed raw) | C |
+| `src/user/.claude/skills/fablize/SKILL.md` | reads sibling closed-bead `notes`/close text; "spec out beads" trigger vocab | RTF | `work list --parent` + `work show`; trigger vocabulary unchanged | C |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in 4c6fcbcc2fca4afc138c184c599829b97c322564. The fablize mapping row now names both queries — work list --parent for open children and work list --parent --status closed for close-note evidence — and acceptance criterion 8 pins both before candidate selection.

**Comment by scotthamilton77** [comment]
Review header; the three inline suggestions are addressed individually — see the thread replies.

**Comment by scotthamilton77** [comment]
@codex review

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+| `src/user/.agents/skills/merge-guard/SKILL.md` | `label list <id> --json \| jq join` | RTF | `work label list` (envelope `string[]`, no jq) | B |
+| `src/user/.agents/skills/wait-for-pr-comments/SKILL.md` | `label add <id> human`, `update --append-notes`, `label list --json`; `.beads/worker-audit/` path | RTF (path=BDS) | `work label add`, `work note`, `work label list`; `.beads/` audit path stays bd infra | B |
+| `src/user/.agents/skills/reply-and-resolve-pr-threads/SKILL.md` | `label add <id> human` + `update --append-notes`; "no bd jargon in replies" policy | RTF (policy=BDS) | `work label add`, `work note`; PR-hygiene policy unaffected (output rule, not a call) | B |
+| `src/user/.agents/skills/triaging-discovered-work/SKILL.md` | `create --parent`, `dep add --type discovered-from`, `reopen`, `create` | RTF | the filing recipe targets **`work discover`** (the composed verb — see the 2026-07-17 work-discover-verb spec), NOT raw `work create` + `work dep add`, which would reinstate the unanchored-filing leak the verb closes; `work reopen` and reads migrate directly (typed dep edges verified shipped end-to-end if ever needed raw) | C |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md:64]: The work-discover verb spec is not missing — it ships in sibling PR #301 (docs/specs/2026-07-17-work-discover-verb.md), open alongside this PR, and the Class C slice is explicitly sequenced behind that verb landing. The reference resolves once both merge; the review necessarily cannot see the sibling branch.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+- **bd remains for humans.** Raw `bd` stays available for direct human use and for the
+  stays-bd-specific teaching docs; the migration removes bd from *asset call paths* only.
+
+## 7. Acceptance criteria (mechanically checkable)
+
+1. `grep -rn 'bd ' <Class A/B/C asset paths>` returns only lines carrying a `facade-gap:`
+   annotation (with G1/G2 verified covered, the expected count is zero; the policy
+   remains for any gap a migration child uncovers).
+2. Every migrated call site invokes `work <verb>` and parses the `{ok,data,error}`
+   envelope — no residual `--json | jq` or `--limit 0` in Class A/B/C.
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in ece3e00ca7c20c41a539c89443ba482aed83c82f. The prohibition is now scoped to --limit 0 on facade/bd invocations only; collect.py's public --limit rendering flag is explicitly exempted and survives the migration unchanged.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+| `src/user/.agents/skills/triaging-discovered-work/SKILL.md` | `create --parent`, `dep add --type discovered-from`, `reopen`, `create` | RTF | the filing recipe targets **`work discover`** (the composed verb — see the 2026-07-17 work-discover-verb spec), NOT raw `work create` + `work dep add`, which would reinstate the unanchored-filing leak the verb closes; `work reopen` and reads migrate directly (typed dep edges verified shipped end-to-end if ever needed raw) | C |
+| `src/user/.claude/skills/fablize/SKILL.md` | reads sibling closed-bead `notes`/close text; "spec out beads" trigger vocab | RTF | `work list --parent <id>` (open children) **and** `work list --parent <id> --status closed` (close-note supersession evidence — bd/facade default omits closed items), each read via `work show`; trigger vocabulary unchanged | C |
+| `src/user/.claude/skills/orchestrating-subagents/SKILL.md` | prose "verify git/gh/bd first" | RTF (light) | swap bd→`work show` in the verify-state noun | C |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in d8c726bcbd2c2a78c1220610eec47982bc1fc0b6. The mapping row now gives a concrete selection procedure: work show with the worker's claimed item id from the dispatch brief, falling back to work list --status in_progress when the id is unknown.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+   concept — the rationale is recorded in the mapping table, not necessarily in the asset.
+
+## 3. Mapping table (central artifact)
+
+Classification is per **file** by dominant usage. `RTF` = route-through-facade,
+`BDS` = stays-bd-specific, `GAP` = facade-gap (partial).
+
+| asset (path) | bd usages (summarized) | class | target work verb(s) / gap | asset-class |
+|---|---|---|---|---|
+| `src/user/.agents/skills/whats-next/collect.py` | `ready --json --limit 0`, `list --status open,in_progress --limit 0`, `list --label`, `list --parent`, `show <id> --json`; parses `id/parent/labels/notes/started_at/assignee` | RTF (+verify G2,G3) | `work ready`, `work list [--status --label --parent]`, `work show`; envelope replaces jq/`--limit 0` | A |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in 832a6a787ba85adca4e4af3b3a92aa3650c002fd. The typed+ready planning queries are now in the collect.py mapping row, and a new facade gap G4 (a --ready flag on work list) is registered and threaded through the partition, ordering, acceptance criterion 6, and the gap continuation.

**Comment by scotthamilton77** [comment]
Review header; three findings fixed and one answered in its thread — see the thread replies.

**Comment by scotthamilton77** [comment]
@codex review

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+| `src/user/.agents/skills/merge-guard/SKILL.md` | `label list <id> --json \| jq join` | RTF | `work label list` (envelope `string[]`, no jq) | B |
+| `src/user/.agents/skills/wait-for-pr-comments/SKILL.md` | `label add <id> human`, `update --append-notes`, `label list --json`; `.beads/worker-audit/` path | RTF (path=BDS) | `work label add`, `work note`, `work label list`; `.beads/` audit path stays bd infra | B |
+| `src/user/.agents/skills/reply-and-resolve-pr-threads/SKILL.md` | `label add <id> human` + `update --append-notes`; "no bd jargon in replies" policy | RTF (policy=BDS) | `work label add`, `work note`; PR-hygiene policy unaffected (output rule, not a call) | B |
+| `src/user/.agents/skills/triaging-discovered-work/SKILL.md` | `create --parent`, `dep add --type discovered-from`, `reopen`, `create` | RTF | the filing recipe targets **`work discover`** (the composed verb — see the 2026-07-17 work-discover-verb spec), NOT raw `work create` + `work dep add`, which would reinstate the unanchored-filing leak the verb closes; `work reopen` and reads migrate directly (typed dep edges verified shipped end-to-end if ever needed raw) | C |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md:64]: The work-discover spec ships in sibling PR #301, open alongside this PR, and its Continuations section carries the concrete implementation work item for the verb — that spec owns the verb contract and its continuation, so duplicating the continuation here would create two competing sources. The Class C slice remains sequenced behind that verb landing.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+  timestamp nor an assignee; `collect.py`'s in-flight section reads bd's `started_at`
+  (for claim age) **and** bd's `assignee` (emitted per in-flight item). Both are absent
+  from `Item` and its serialized envelope, so a naive migration would render every
+  assigned in-flight item as unassigned and lose claim age. Resolution: two additive
+  `Item` fields sourced from bd — `started` (from `started_at`, `None` when unset) and
+  `assignee` (from bd's `assignee`, `None`/empty when unset) — one MINOR protocol bump.
+  Filed as a single continuation. Additionally, the Class A migration must confirm the
+  normalized `type`/`created` fields remap correctly onto `collect.py`'s consumers (bd's
+  raw `created_at`/type strings are renamed on the `Item` boundary), so the in-flight
+  and ready sections read the envelope's normalized names, not the raw bd keys.
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in e980b9d363bbede058b993db133734ea310c38b0. The Class A remapping note now requires converting the facade's P<n> priority string back to the integer at the Item boundary, preserving sort order and the public JSON shape.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+  `collect.py`'s own public `--limit N` CLI flag (max beads per rendered section,
+  `0 = show everything`), which is a rendering option, not a backend call.
+- **bd remains for humans.** Raw `bd` stays available for direct human use and for the
+  stays-bd-specific teaching docs; the migration removes bd from *asset call paths* only.
+
+## 7. Acceptance criteria (mechanically checkable)
+
+1. `grep -rn 'bd ' <Class A/B/C asset paths>` returns only lines carrying a `facade-gap:`
+   annotation (with G1/G2 verified covered, the expected count is zero; the policy
+   remains for any gap a migration child uncovers).
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md]: Fixed in 9a72368205c8c6b375200db489067cebc36a68cd. Acceptance criterion 1 now excludes test-fixture files (evals/, *_test.*) from the raw-bd grep — fixture prompts are test data, not call sites, and need no annotation.

**Comment by scotthamilton77** [comment]
Review header; two findings fixed, one answered — see the thread replies.

**Comment by scotthamilton77** [comment]
@codex review

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+| `src/user/.agents/skills/merge-guard/SKILL.md` | `label list <id> --json \| jq join` | RTF | `work label list` (envelope `string[]`, no jq) | B |
+| `src/user/.agents/skills/wait-for-pr-comments/SKILL.md` | `label add <id> human`, `update --append-notes`, `label list --json`; `.beads/worker-audit/` path | RTF (path=BDS) | `work label add`, `work note`, `work label list`; `.beads/` audit path stays bd infra | B |
+| `src/user/.agents/skills/reply-and-resolve-pr-threads/SKILL.md` | `label add <id> human` + `update --append-notes`; "no bd jargon in replies" policy | RTF (policy=BDS) | `work label add`, `work note`; PR-hygiene policy unaffected (output rule, not a call) | B |
+| `src/user/.agents/skills/triaging-discovered-work/SKILL.md` | `create --parent`, `dep add --type discovered-from`, `reopen`, `create` | RTF | the filing recipe targets **`work discover`** (the composed verb — see the 2026-07-17 work-discover-verb spec), NOT raw `work create` + `work dep add`, which would reinstate the unanchored-filing leak the verb closes; `work reopen` and reads migrate directly (typed dep edges verified shipped end-to-end if ever needed raw) | C |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md:64]: The verb is not a facade gap in this spec's sense — its full contract spec ships in sibling PR #301 (open alongside this PR) whose Continuations own the implementation work item, and the Class C slice is explicitly sequenced behind that verb landing. The gap register tracks capabilities with NO owned design; this one has an owned design one PR over.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+   concept — the rationale is recorded in the mapping table, not necessarily in the asset.
+
+## 3. Mapping table (central artifact)
+
+Classification is per **file** by dominant usage. `RTF` = route-through-facade,
+`BDS` = stays-bd-specific, `GAP` = facade-gap (partial).
+
+| asset (path) | bd usages (summarized) | class | target work verb(s) / gap | asset-class |
+|---|---|---|---|---|
+| `src/user/.agents/skills/whats-next/collect.py` | `ready --json --limit 0`, `list --status open,in_progress --limit 0`, `list --label`, `list --parent`, three `list --type {milestone,epic,feature} --ready` (planning queue), `show <id> --json`; parses `id/parent/labels/notes/started_at/assignee` | RTF (+verify G2; gaps G3,G4) | `work ready`, `work list [--status --label --parent]`, `work show`; planning-queue typed+ready query needs G4; envelope replaces jq/`--limit 0` | A |
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md:57]: Covered by the Class A remapping requirement: the facade's show returns a single typed item envelope, which removes the bd single-element-array [0] indexing rather than requiring new normalization — a simplification the remapping note's type/created/priority list already exemplifies; the migration child's review verifies the full field map.

File: docs/specs/2026-07-17-bd-to-work-facade-migration.md [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+  `collect.py`'s own public `--limit N` CLI flag (max beads per rendered section,
+  `0 = show everything`), which is a rendering option, not a backend call.
+- **bd remains for humans.** Raw `bd` stays available for direct human use and for the
+  stays-bd-specific teaching docs; the migration removes bd from *asset call paths* only.
+
+## 7. Acceptance criteria (mechanically checkable)
+
+1. `grep -rn 'bd ' <Class A/B/C asset paths>`, **excluding test-fixture files
+   (`evals/`, `*_test.*`)**, returns only lines carrying a `facade-gap:`
+   annotation (with G1/G2 verified covered, the expected count is zero; the policy
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply docs/specs/2026-07-17-bd-to-work-facade-migration.md:173]: In Class A/B/C, prose bd recipes ARE the migration surface — they are instruction call sites an agent executes, so the grep flagging them is intended, and stays-bd-specific teaching prose lives in Class D/E, which the criterion does not grep. The fixture exemption already covers non-executable test data.

**Comment by scotthamilton77** [comment]
Review header; every inline finding is answered in its thread (edge-case semantics recorded on the implementation work item, the rest rebutted with rationale).
